pkg/util/utils_linux.go

```diff
@@ -258,3 +265,70 @@ func DeviceFromPath(path string) (*spec.LinuxDevice, error) {
 		Minor:    int64(unix.Minor(devNumber)),
 	}, nil
 }
+
+// kernelSupportsRecursivelyReadOnly returns true if the kernel supports recursive readonly mounts
+// from https://github.com/moby/moby/blob/master/daemon/daemon_linux.go#L222
+func kernelSupportsRecursivelyReadOnly() error {
+	fn := func() error {
+		tmpMnt, err := os.MkdirTemp("", "podman-detect-rro")
+		if err != nil {
+			return fmt.Errorf("failed to create a temp directory: %w", err)
+		}
+		for {
+			err = unix.Mount("", tmpMnt, "tmpfs", 0, "")
+			if !errors.Is(err, unix.EINTR) {
+				break
+			}
+		}
+		if err != nil {
+			return fmt.Errorf("failed to mount tmpfs on %q: %w", tmpMnt, err)
+		}
+		defer func() {
+			var umErr error
+			for {
+				umErr = unix.Unmount(tmpMnt, 0)
+				if !errors.Is(umErr, unix.EINTR) {
+					break
+				}
+			}
+			if umErr != nil {
+				logrus.Errorf("Failed to unmount %q: %v", tmpMnt, umErr)
+			}
+		}()
```

Comment:
please use `open_tree` or `fsopen` so we don't have to create a mount that could be leaked to the host.